src/utility.c

```diff
@@ -471,3 +471,171 @@ int pathcmp(const char *a, const char *b) {
     free(myB);
     return ret;
 }
+
+int is_json_array(const char *value) {
+  if (value[0]=='[' && value[1]=='u' && value[2]=='\''){
+    return 1;
+  }
+  else {
+    return 0;
+  }
+}
+
+/**
+ * split_json_array - utility function to parse a JSON array and split it
+ * into an array of strings.  It will allocate the array but destroys the
+ * source string.
+ *
+ * The values in the array should be comma-separated quoted strings.
+ */
+char **split_json_array(const char *value) {
```

Comment:
this function assumes that is_json_array has been called externally, and then manipulates the buffers as such.

is_json_array should be made static and this function should be the only caller for it; this function should internally handle the memory instead of the external duplicates.

Reply:
It isn't depending on is_json_array.  Note that it does a strndup on line 549.  So it is making its own copy.